tests need python modules that are not installed yet: so point the built ones.
Reported upstream: https://bugreports.qt.io/browse/PYSIDE-1429

--- a/sources/shiboken2/tests/CMakeLists.txt	2020-09-09 14:45:42.000000000 +0200
+++ b/sources/shiboken2/tests/CMakeLists.txt	2020-11-12 14:16:46.577206385 +0100
@@ -53,6 +53,6 @@
     list(FIND test_blacklist ${test_name} expect_fail)
     add_test(${test_name} ${PYTHON_EXECUTABLE} ${test_file})
-    set_tests_properties(${test_name} PROPERTIES ENVIRONMENT "BUILD_DIR=${BUILD_DIR}")
+    set_tests_properties(${test_name} PROPERTIES ENVIRONMENT "BUILD_DIR=${BUILD_DIR};PYTHONPATH=${shibokenmodule_BINARY_DIR}:${minimal_BINARY_DIR}:${sample_BINARY_DIR}:${other_BINARY_DIR}:${smart_BINARY_DIR}")
     set_tests_properties(${test_name} PROPERTIES TIMEOUT ${CTEST_TESTING_TIMEOUT})
     if (${expect_fail} GREATER -1)
         set_tests_properties(${test_name} PROPERTIES WILL_FAIL TRUE)
